### PR TITLE
Close doc preview on release merge

### LINKED ISSUE #76: Merge cobradoc-preview PRs after corresponding Vitess PR is merged
Right now, the bot handles 3 different scenarios for cobradocs:

1. When a PR is `opened` or `pushed` against `vitess`, the bot opens (or updates) a PR against `website` to provide a preview of the website doc changes caused by that vitess PR. This PR is flagged with a big "[DO NOT MERGE]" label, and is meant to be discarded after the original vitess PR is merged.
2. When a PR is `closed` against `vitess`:
    a. If merged into `main`, [update any cobradoc changes](https://github.com/vitessio/vitess-bot/blob/main/go/pull_request.go#L151-L161).
    b. If merged into a branch other than `main` (i.e. `release-18.0`), [do nothing](https://github.com/vitessio/vitess-bot/blob/main/go/pull_request.go#L725-L728).
3. When a release is tagged, generate all cobradoc changes for all versions. This handles the case of any doc changes in 2b we discarded.

It would be better if we instead combined scenarios 1 and 2a, so that when a PR is merged against `vitess:main`, we take the original "[DO NOT MERGE]" preview PR against `website`, synchronize the docs one more time, push, and automerge it.

Additionally, when we merge a PR against a `vitess` branch other than `main` (ie. scenario 2b), we close the `website` PR we opened in scenario 1.

### PR DESCRIPTION
**Pull request by ajm188**
Tested with https://github.com/ajm188/website/pull/17

Some relevant log lines from the testing:

```
{"level":"info","github_event_type":"pull_request","github_delivery_id":"e6283e60-bf75-11ee-958a-25db88c0abee","github_installation_id":42461852,"github_repository_owner":"ajm188","github_repository_name":"vitess","github_pr_num":14,"time":"2024-01-30T09:54:15-05:00","message":"No cobradocs changed for PR https://github.com/ajm188/vitess.git/vitess#14 at base release-18.0. Skipping first commit ..."}
```

```
{"level":"info","github_event_type":"pull_request","github_delivery_id":"38f51910-bf80-11ee-942a-1075829bfa8d","github_installation_id":42461852,"github_repository_owner":"ajm188","github_repository_name":"vitess","github_pr_num":14,"time":"2024-01-30T09:59:54-05:00","message":"closing open PR ajm188/website#17"}
```

Closes #76

### CODE REVIEW

File: go/pull_request.go [hunk 1]
```diff
@@ -803,13 +828,46 @@ func (h *PullRequestHandler) updateDocs(ctx context.Context, event github.PullRe
 		}
 	}()
 
+	website := git.NewRepo(
```
Comment (ajm188) [inline go/pull_request.go:831]: this needed to be moved up to be referenced in `website.FindPRs`

File: go/pull_request.go [hunk 2]
```diff
@@ -628,30 +629,54 @@ func (h *PullRequestHandler) createCobraDocsPreviewPR(
...
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		tree = &github.Tree{
+			SHA: &baseTree,
+		}
+		commit = &github.Commit{
+			SHA: &parent,
+		}
```
Comment (ajm188) [inline go/pull_request.go:679]: these are here so the calls in the second pass of `writeAndCommitTree` are set up correctly to point to the original base/parent tree/commit

**Comment by frouioui** [comment]
The log `Skipping first commit ...` was for the first commit on your PR that is just a merge?

File: go/pull_request.go [hunk 3]
```diff
@@ -628,30 +629,54 @@ func (h *PullRequestHandler) createCobraDocsPreviewPR(
...
 	_, err = shell.NewContext(ctx, "./tools/sync_cobradocs.sh").InDir(website.LocalDir).WithExtraEnv(
 		fmt.Sprintf("VITESS_DIR=%s", vitess.LocalDir),
 		"COBRADOCS_SYNC_PERSIST=yes",
 		fmt.Sprintf("COBRADOC_VERSION_PAIRS=HEAD:%s", docsVersion),
 	).Output()
 	if err != nil {
-		return nil, errors.Wrapf(err, "Failed to run cobradocs sync script against %s:%s to %s for %s", remote, ref, op, pr.GetHTMLURL())
+		var exitErr *exec.ExitError
+		if (errors.As(err, &exitErr)) &&
+			(bytes.Contains(exitErr.Stderr, []byte("No changes to cobradocs detected"))) {
```
Comment (frouioui) [inline go/pull_request.go:647]: nit picking but there are un-required parenthesis on this if condition
```suggestion
		if errors.As(err, &exitErr) &&
			bytes.Contains(exitErr.Stderr, []byte("No changes to cobradocs detected")) {
```

**Review by frouioui — APPROVED** [review]

**Comment by ajm188** [comment]
> The log Skipping first commit ... was for the first commit on your PR that is just a merge?

The first commit (i.e. base branch) of the vitess PR, which would either be `main` or a `release-` branch

**Pull request merged**